### DISCUSSION TITLE
Read-only sqlite

### DISCUSSION THREAD

**Discussion by jpcafe10** [vercel/community#1181] [Help] [answered]
Hi,

I see this question asked many times but with no clear solution/explanation.

Deployed read-only sqlite on vercel, is it possible? If so, how?

The idea is to keep the .db in the repo and publish updates on git push. I’m currently using Postgres but I could save some ££ and querying would be faster.

**Comment by jpcafe10** [comment]
Forgot to say, data changes every 3 months so read only sqlite would be perfect for my use case.

**Reply by cademis** [reply]
I had a similar use case for a read only sqlite. I used the method described in the link below and now the deployed app has full read only access to the db

https://github.com/keystonejs/keystone/discussions/8990#discussioncomment-8984266

**Comment by amyegan** [comment]
Hi @jca41! You won't be able to host sqlite on Vercel ([more info](https://vercel.com/guides/is-sqlite-supported-in-vercel)), but there are other options. You can use one of the available [database integrations](https://vercel.com/integrations#databases) to connect a free database host if this is a relatively small project.

**Reply by nileshtrivedi** [reply]
The link you shared IS the source of this confusion. It says that ephemeral storage is available in Vercel. So it should be possible to use a sqlite db file for read-only operations? Each serverless instance will have its own copy but that shouldn't be a problem.

**Reply by amyegan** [reply, marked as answer]
Somehow I missed it before, but I just found a demo repo with a read-only sqlite example! There are some [limitations](https://vercel.com/docs/concepts/functions/serverless-functions#limits) that make hosting a database in a serverless function impractical, but maybe this will work for you: https://github.com/WebReflection/vercel-sqlite-demo

**Comment by phimong** [comment]
It would be perfect if Vercel could support SQLITE reads
